internal/pkg/v2/README.md

```diff
@@ -0,0 +1,296 @@
+
+
+
+### Approach
+
+The scope of APIv2 prevents complete implementation across all services within a single EdgeX release cycle.  The community has decided to release Geneva with APIv1 and "experimental/beta" APIv2 support for  some subset of services.  The intention is for the future EdgeX 2.0 release to include only APIv2 endpoints.
+
+APIv2 is being treated as greenfield development.  This reduces risk by minimizing overlap with APIv1 implementation.  It ensures APIv2 is able to be developed and evolve separately from APIv1.  It also provides a clear delineation between APIv1 and APIv2 and makes it easier (and less likely to introduce last-minute issues) to remove the APIv1 functionality for the future EdgeX 2.0 release.
+
+To this end, the community decided that APIv2 persistence will be separate from APIv1 persistence.  This supports independent schema development at the expense of data access across the two API versions.  That is, data persisted via APIv1 will not be available via APIv2 (and vice-versa).
+
+
+
+### Definitions
+
+- ***Use-case:*** discrete behavior initiated by external interaction.  The external interaction could be a user or another service.  A use case is associated with a version, a type, and an action.
+- ***Version:*** the major value of the semantic version of a specific use- case implementation (e.g. "2").
+- ***Type:*** a mnemonic associated with a specific use-case implementation (e.g. "ping").
+- ***Action:*** a generic (HTTP method-equivalent) name associated with a specific use-case implementation; currently defined are "create", "read", "update", "delete", and "command."
+- ***DTO:*** data transfer object.  An object with no behavior used only for data transmission; e.g. the request and response objects that define the structure and content of the data required by and returned by a use-case.
+- ***Use-case Request:*** a request DTO instance that maps to a single invocation of a use-case.
+- ***Use-case Response:*** a response DTO instance returned from a single invocation of a use-case. 
+- ***Transport Request:***  in the current implementation, this is a single HTTP request.  A single transport request can contain one or more use-case requests and an equivalent number of use-case responses.
+- ***Use-case Endpoint:*** URL that accepts one or more use-case-specific requests and returns an equivalent number of use-case responses.  
+- ***Batch Endpoint:*** un-versioned URL (`/api/batch`) that accepts one or more use-case requests and returns an equivalent number of use-case responses. 
+
+
+
+### Layered Architecture
+
+The implementation is based on the layers common to a domain-driven design application that uses traditional Layers Architecture (see Buschmann, Frank, et al. 1996. *Pattern-Oriented Software Architecture, Volume 1: A System of Patterns*. New York: Wiley).
+
+The implementation spans 4 architectural layers:
+
+- ***User Interface Layer***  
```

Comment:
I really dislike this term... I understand it's historical, but there's nothing "user interface" about this design.

Reply:
https://github.com/edgexfoundry/edgex-go/pull/2285#discussion_r373254118

Unresolve if you want to revisit/continue this conversation.